jobs: TestRegistryGC failed

jobs.TestRegistryGC [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/12563530?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/12563530?buildTab=artifacts#/) on master @ [cd3d2e3c7a7356c7e0604f32ff3cb4e88ed12845](https://github.com/cockroachdb/cockroach/commits/cd3d2e3c7a7356c7e0604f32ff3cb4e88ed12845):


```
=== RUN   TestRegistryGC
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/c70a284aa5d459208ac4177c50bcf8ed/logTestRegistryGC3828296434
    test_log_scope.go:81: use -show-logs to present logs inline
    test_server_shim.go:146: cluster virtualization disabled in global scope due to issue: #76378 (expected label: C-bug)
    conditional_wrap.go:190: 
        pkg/jobs/registry_test.go:246: (TestRegistryGC)
        	NOTICE: .JobRegistry() called via implicit interface ApplicationLayerInterface;
        HINT: consider using .ApplicationLayer().JobRegistry() instead.
    conditional_wrap.go:194: TIP: consider replacing the test server initialization from:
            ts, ... := serverutils.StartServer(t, ...)
            defer ts.Stopper().Stop(...)
        to:
            srv, ... := serverutils.StartServer(t, ...)
            defer srv.Stopper().Stop(...)
            ts := srv.ApplicationLayer()
        
        See also: https://go.crdb.dev/p/testserver-and-cluster-virtualization
    registry_test.go:247: TransactionRetryWithProtoRefreshError: ReadWithinUncertaintyIntervalError: read at time 1699427798.139023712,0 encountered previous write with future timestamp 1699427798.139023712,1 (local=1699427798.138167750,1) within uncertainty interval ``t <= (local=1699427798.139023712,0, global=1699427798.639023712,0)``; observed timestamps: [{1 1699427798.139023712,0}]: "sql txn" meta={id=2fbf5866 key=/Min iso=Serializable pri=0.00282438 epo=0 ts=1699427798.139023712,0 min=1699427798.139023712,0 seq=0} lock=false stat=PENDING rts=1699427798.139023712,0 wto=false gul=1699427798.639023712,0
    panic.go:523: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/c70a284aa5d459208ac4177c50bcf8ed/logTestRegistryGC3828296434
--- FAIL: TestRegistryGC (4.20s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/jobs @cockroachdb/disaster-recovery
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRegistryGC.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-33294

--- a/pkg/jobs/registry_test.go
+++ b/pkg/jobs/registry_test.go
@@ -115,7 +115,7 @@ func TestRegistryGC(t *testing.T) {
 	defer log.Scope(t).Close(t)
 
 	ctx := context.Background()
-	s, sqlDB, kvDB := serverutils.StartServer(t, base.TestServerArgs{
+	srv, sqlDB, kvDB := serverutils.StartServer(t, base.TestServerArgs{
 		Knobs: base.TestingKnobs{
 			SpanConfig: &spanconfig.TestingKnobs{
 				// This test directly modifies `system.jobs` and makes over its contents
@@ -137,7 +137,8 @@ func TestRegistryGC(t *testing.T) {
 			JobsTestingKnobs: NewTestingKnobsWithShortIntervals(),
 		},
 	})
-	defer s.Stopper().Stop(ctx)
+	defer srv.Stopper().Stop(ctx)
+	s := srv.ApplicationLayer()
 
 	db := sqlutils.MakeSQLRunner(sqlDB)
 
@@ -243,9 +244,13 @@ INSERT INTO t."%s" VALUES('a', 'foo');
 				{sqlActivityJob}, {oldRunningJob}, {oldSucceededJob}, {oldFailedJob}, {oldRevertFailedJob}, {oldCanceledJob},
 				{newRunningJob}, {newSucceededJob}, {newFailedJob}, {newRevertFailedJob}, {newCanceledJob}})
 
-			if err := s.JobRegistry().(*Registry).cleanupOldJobs(ctx, earlier); err != nil {
-				t.Fatal(err)
-			}
+			testutils.SucceedsSoon(t, func() error {
+				if err := s.JobRegistry().(*Registry).cleanupOldJobs(ctx, earlier); err != nil {
+					return err
+				}
+				return nil
+			})
+
 			db.CheckQueryResults(t, `SELECT id FROM system.jobs ORDER BY id`, [][]string{
 				{sqlActivityJob}, {oldRunningJob}, {oldRevertFailedJob}, {newRunningJob},
 				{newSucceededJob}, {newFailedJob}, {newRevertFailedJob}, {newCanceledJob}})
@@ -269,7 +274,7 @@ func TestRegistryGCPagination(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
 	ctx := context.Background()
-	s, sqlDB, _ := serverutils.StartServer(t, base.TestServerArgs{
+	srv, sqlDB, _ := serverutils.StartServer(t, base.TestServerArgs{
 		Knobs: base.TestingKnobs{
 			SpanConfig: &spanconfig.TestingKnobs{
 				// This test directly modifies `system.jobs` and makes over its contents
@@ -293,7 +298,8 @@ func TestRegistryGCPagination(t *testing.T) {
 		},
 	})
 	db := sqlutils.MakeSQLRunner(sqlDB)
-	defer s.Stopper().Stop(ctx)
+	defer srv.Stopper().Stop(ctx)
+	s := srv.ApplicationLayer()
 
 	for i := 0; i < 2*cleanupPageSize+1; i++ {
 		payload, err := protoutil.Marshal(&jobspb.Payload{})

